src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py

```diff
@@ -170,9 +188,34 @@ def _create_results_validator(self):
         self.results_validator = SimulationResultValidator(is_scm=False)
 
     @staticmethod
-    def _create_endpoint_buffer(should_export_plots):
+    def _create_results_handler(should_export_plots):
         return ResultsHandler(should_export_plots)
 
+    def create_hierarchy_stats(self, area: "AreaBase"):
+        """Calculate hierarchy related statistics."""
+        hierarchy_self_consumption_list = {}
+        self._calc_hierarchy_stats(area, 0, hierarchy_self_consumption_list)
+        for level, consumption_list in hierarchy_self_consumption_list.items():
```

Comment:
The KPI value is not the intended one here. We actually want to replicate this:
![image](https://github.com/user-attachments/assets/5cafc411-99f9-4bdb-bdb4-291508cb2074)

Therefore I think that this can be done a bit more elaborate here. Let's discuss it on Slack to settle. I will also add the screenshot in the ticket in order to have it available.